doc/audit/merkle_tree.rst

```diff
@@ -1,14 +1,30 @@
 Merkle Tree
 ===========
 
-The high-integrity guarantees of CCF are enforced by a single :term:`Merkle Tree` which records all transactions that mutate the key-value store. The root of the Merkle Tree is regularly signed by the primary node and the signature is recorded in the ``ccf.signatures`` key-value map. Like any other transaction, the signature transaction is also recorded in the ledger, which allows for offline auditability of the service.
-
 .. note:: The :ref:`CCF Python package <quickstart/install:Python Package>` provides the ``ledger`` module to easily parse the ledger. More details :doc:`here <python_library>`.
 
-The signature transaction also contains the Merkle Tree so far (note: the tree is compacted at regular interval to avoid infinite memory growth).
+The high-integrity guarantees of CCF are enforced by a single :term:`Merkle Tree` which records the cryptographic hash (leaves) of all transactions that mutate the key-value store. The root of the Merkle Tree is regularly signed by the primary node and the signature is recorded in the ``public:ccf.internal.signatures`` key-value map. Like any other transaction, the signature transaction is also recorded in the ledger, which allows for offline auditability of the service.
+
+The following diagram demonstrates how the integrity of the ledger can be verified: a signature transaction (at ``seqno`` 6) signs the root of the Merkle Tree so far (after ``seqno`` 5).
+
+.. image:: ../img/merkle_single_signature.svg
+  :width: 600
+  :align: center
+
+Auditors of the CCF ledger can reconstruct the Merkle Tree by walking through the ledger, hashing the transactions recorded in the ledger and appending them to a fresh Merkle Tree. On every signature transaction, the reconstituted root can be compared to the one recorded in the signature transaction. The signature can be verified using the public signing key of the primary node at the time (also recorded in the ledger -- see ``public:ccf.gov.nodes`` map). If the signature and root match, the integrity of the ledger so far is guaranteed.
+
+.. note:: The signature transaction also contains the Merkle Tree so far. This allows CCF to efficiently issue receipts for historical transactions by only reading the signature that follows the target transaction in the ledger.
```

Comment:
"transaction also contains the Merkle Tree": the signature doesn't actually contain a copy of the entire tree, maybe better "the tree segment of the signed transaction"?

Reply:
Good point. I've re-phrased to "The signature transaction also contains the hash of the transactions signed since the last signature" as I didn't want to introduce the word "segment". Happy to be convinced otherwise.

Reply:
Sure, that sounds good!